clang/lib/Format/UnwrappedLineFormatter.cpp

```diff
@@ -1355,8 +1355,11 @@ unsigned UnwrappedLineFormatter::format(
     bool FixIndentation = (FixBadIndentation || ContinueFormatting) &&
                           Indent != TheLine.First->OriginalColumn;
     bool ShouldFormat = TheLine.Affected || FixIndentation;
-    if (Style.IgnorePPDefinitions && TheLine.Type == LT_PreprocessorDirective)
+    if (Style.IgnorePPDefinitions && TheLine.Type == LT_PreprocessorDirective &&
+        TheLine.getFirstNonComment()->Next->is(tok::pp_define)) {
```

Comment:
Does `Next` always exist? Is just `#` marked as `LT_PreprocessorDirective`?

Reply:
Good point, better safe than sorry. I've refactored the condition in the next changeset.